This patch was written by Yu Shao <yshao@redhat.com>, and applies on top of
a previous patch of work done by James Su <suzhe@turbolinux.com.cn> and Yu
Shao in patch:  XFree86-4.3.0-gb18030-20020207.patch which was formerly
patch XFree86-4.2.0-gb18030-20020207.patch in Red Hat Linux, but lacked
proper credits.

Mike A. Harris <mharris@redhat.com>

diff -uNr xc.orig/fonts/encodings/large/gb18030.2000-1.enc xc/fonts/encodings/large/gb18030.2000-1.enc
--- xc.orig/fonts/encodings/large/gb18030.2000-1.enc	Fri Mar 15 23:17:16 2002
+++ xc/fonts/encodings/large/gb18030.2000-1.enc	Fri Mar 15 23:19:08 2002
@@ -2,7 +2,220 @@
 SIZE 0x100 0x100
 STARTMAPPING unicode
 UNDEFINE 0x80 0xFFFF
-0x80  0xFFFF  0x80
+0x00a7  0x00a8  0x00a7
+0x00b0  0x00b1  0x00b0
+0x00b7  0x00b7
+0x00d7  0x00d7
+0x00e0  0x00e1  0x00e0
+0x00e8  0x00ea  0x00e8
+0x00ec  0x00ed  0x00ec
+0x00f2  0x00f3  0x00f2
+0x00f7  0x00f7
+0x00f9  0x00fa  0x00f9
+0x00fc  0x00fc
+0x0101  0x0101
+0x0113  0x0113
+0x011b  0x011b
+0x012b  0x012b
+0x0144  0x0144
+0x0148  0x0148
+0x014d  0x014d
+0x016b  0x016b
+0x01ce  0x01ce
+0x01d0  0x01d0
+0x01d2  0x01d2
+0x01d4  0x01d4
+0x01d6  0x01d6
+0x01d8  0x01d8
+0x01da  0x01da
+0x01dc  0x01dc
+0x01f9  0x01f9
+0x0251  0x0251
+0x0261  0x0261
+0x02c7  0x02c7
+0x02c9  0x02cb  0x02c9
+0x02d9  0x02d9
+0x0391  0x03a1  0x0391
+0x03a3  0x03a9  0x03a3
+0x03b1  0x03c1  0x03b1
+0x03c3  0x03c9  0x03c3
+0x0401  0x0401
+0x0410  0x044f  0x0410
+0x0451  0x0451
+0x060c  0x060c
+0x061b  0x061b
+0x061f  0x061f
+0x0626  0x0628  0x0626
+0x062a  0x062a
+0x062c  0x062f  0x062c
+0x0631  0x0634  0x0631
+0x0639  0x063a  0x0639
+0x0640  0x064a  0x0640
+0x0674  0x0678  0x0674
+0x067e  0x067e
+0x0686  0x0686
+0x0698  0x0698
+0x06a9  0x06a9
+0x06ad  0x06ad
+0x06af  0x06af
+0x06be  0x06be
+0x06c5  0x06c9  0x06c5
+0x06cb  0x06cc  0x06cb
+0x06d0  0x06d0
+0x06d5  0x06d5
+0x0f00  0x0f47  0x0f00
+0x0f49  0x0f6a  0x0f49
+0x0f71  0x0f8b  0x0f71
+0x0f90  0x0f97  0x0f90
+0x0f99  0x0fbc  0x0f99
+0x0fbe  0x0fcc  0x0fbe
+0x0fcf  0x0fcf
+0x1800  0x180e  0x1800
+0x1810  0x1819  0x1810
+0x1820  0x1877  0x1820
+0x1880  0x18a9  0x1880
+0x2010  0x2010
+0x2013  0x2016  0x2013
+0x2018  0x2019  0x2018
+0x201c  0x201d  0x201c
+0x2025  0x2026  0x2025
+0x2030  0x2030
+0x2032  0x2033  0x2032
+0x2035  0x2035
+0x203b  0x203b
+0x20ac  0x20ac
+0x2103  0x2103
+0x2105  0x2105
+0x2109  0x2109
+0x2116  0x2116
+0x2121  0x2121
+0x2160  0x216b  0x2160
+0x2170  0x2179  0x2170
+0x2190  0x2193  0x2190
+0x2196  0x2199  0x2196
+0x2208  0x2208
+0x220f  0x220f
+0x2211  0x2211
+0x2215  0x2215
+0x221a  0x221a
+0x221d  0x2220  0x221d
+0x2223  0x2223
+0x2225  0x2225
+0x2227  0x222b  0x2227
+0x222e  0x222e
+0x2234  0x2237  0x2234
+0x223d  0x223d
+0x2248  0x2248
+0x224c  0x224c
+0x2252  0x2252
+0x2260  0x2261  0x2260
+0x2264  0x2267  0x2264
+0x226e  0x226f  0x226e
+0x2295  0x2295
+0x2299  0x2299
+0x22a5  0x22a5
+0x22bf  0x22bf
+0x2312  0x2312
+0x2460  0x2469  0x2460
+0x2474  0x249b  0x2474
+0x2500  0x254b  0x2500
+0x2550  0x2573  0x2550
+0x2581  0x258f  0x2581
+0x2593  0x2595  0x2593
+0x25a0  0x25a1  0x25a0
+0x25b2  0x25b3  0x25b2
+0x25bc  0x25bd  0x25bc
+0x25c6  0x25c7  0x25c6
+0x25cb  0x25cb
+0x25ce  0x25cf  0x25ce
+0x25e2  0x25e5  0x25e2
+0x2605  0x2606  0x2605
+0x2609  0x2609
+0x2640  0x2640
+0x2642  0x2642
+0x2e81  0x2e81
+0x2e84  0x2e84
+0x2e88  0x2e88
+0x2e8b  0x2e8c  0x2e8b
+0x2e97  0x2e97
+0x2ea7  0x2ea7
+0x2eaa  0x2eaa
+0x2eae  0x2eae
+0x2eb3  0x2eb3
+0x2eb6  0x2eb7  0x2eb6
+0x2ebb  0x2ebb
+0x2eca  0x2eca
+0x2ff0  0x2ffb  0x2ff0
+0x3000  0x3003  0x3000
+0x3005  0x3017  0x3005
+0x301d  0x301e  0x301d
+0x3021  0x3029  0x3021
+0x303e  0x303e
+0x3041  0x3093  0x3041
+0x309b  0x309e  0x309b
+0x30a1  0x30f6  0x30a1
+0x30fc  0x30fe  0x30fc
+0x3105  0x3129  0x3105
+0x3220  0x3229  0x3220
+0x3231  0x3231
+0x32a3  0x32a3
+0x338e  0x338f  0x338e
+0x339c  0x339e  0x339c
+0x33a1  0x33a1
+0x33c4  0x33c4
+0x33ce  0x33ce
+0x33d1  0x33d2  0x33d1
+0x33d5  0x33d5
+0x3400  0x4db5  0x3400
+0x4e00  0x9fa5  0x4e00
+0xa000  0xa48c  0xa000
+0xa490  0xa4a1  0xa490
+0xa4a4  0xa4b3  0xa4a4
+0xa4b5  0xa4c0  0xa4b5
+0xa4c2  0xa4c4  0xa4c2
+0xa4c6  0xa4c6
+0xe78d  0xe796  0xe78d
+0xe7c7  0xe7c7
+0xe816  0xe818  0xe816
+0xe81e  0xe81e
+0xe826  0xe826
+0xe82b  0xe82c  0xe82b
+0xe831  0xe832  0xe831
+0xe83b  0xe83b
+0xe843  0xe843
+0xe854  0xe855  0xe854
+0xe864  0xe864
+0xf92c  0xf92c
+0xf979  0xf979
+0xf995  0xf995
+0xf9e7  0xf9e7
+0xf9f1  0xf9f1
+0xfa0c  0xfa0f  0xfa0c
+0xfa11  0xfa11
+0xfa13  0xfa14  0xfa13
+0xfa18  0xfa18
+0xfa1f  0xfa21  0xfa1f
+0xfa23  0xfa24  0xfa23
+0xfa27  0xfa29  0xfa27
+0xfb56  0xfb59  0xfb56
+0xfb7a  0xfb7d  0xfb7a
+0xfb8e  0xfb95  0xfb8e
+0xfbaa  0xfbad  0xfbaa
+0xfbd3  0xfbff  0xfbd3
+0xfe30  0xfe31  0xfe30
+0xfe33  0xfe44  0xfe33
+0xfe49  0xfe52  0xfe49
+0xfe54  0xfe57  0xfe54
+0xfe59  0xfe66  0xfe59
+0xfe68  0xfe6b  0xfe68
+0xfe89  0xfe92  0xfe89
+0xfe95  0xfe98  0xfe95
+0xfe9d  0xfeaa  0xfe9d
+0xfead  0xfeb8  0xfead
+0xfec9  0xfef4  0xfec9
+0xfefb  0xfefc  0xfefb
+0xff01  0xff5e  0xff01
+0xffe0  0xffe5  0xffe0
 ENDMAPPING
 STARTMAPPING cmap 3 4
 # the identity mapping
